### PR TITLE
add generic osx install script.

### PR DESCRIPTION
**Pull request by tsteven4**
This adds a script install-qt-osx modelled on install-qt.

I added travis regression cases to cover this.

I added 5.12.0 to linux regression.  We could drop 5.10.x and 5.11.x from regression now that 5.12.0 LTS has been released.

I have seen rare failures on travis when extract-qt-installer apparently returns before the install is finished causing hdiutil detatch to fail.

### CODE REVIEW

**Comment by van9konst** [comment]
maybe we need to test only LTS versions? no reason to test 5.9.5 and 5.9.6 ...
use only last LTS numbers:
- 5.6.3 
- 5.9.7
- 5.12.0

File: recipes/install-qt-osx [hunk 1]
```diff
@@ -0,0 +1,28 @@
...
+#   https://github.com/musescore/MuseScore/blob/master/build/travis/job_macos/install.sh
+
+set -v
+set -e
+
+QT_VERSION=${1:-5.12.0}
+QT_TARGET_CATALOG=${2:-$PWD}
+QT_CI_DOWNLOADER=${QT_CI_DOWNLOADER:-"wget -c -N"}
+
+DOWNLOAD_URL=https://download.qt.io/archive/qt/$(echo ${QT_VERSION} | cut -d "." -f -2)/${QT_VERSION}/qt-opensource-mac-x64-${QT_VERSION}.dmg
```
Comment (van9konst) [inline recipes/install-qt-osx]: `DOWNLOAD_URL=https://download.qt.io/archive/qt/$(echo ${QT_VERSION} | cut -d "." -f -2)/${QT_VERSION}/qt-opensource-${$3:-linux}-x64-${QT_VERSION}.dmg`

i don't think we need the copy of original "recipes/install-qt". we can just parametrize script as 3rd or parse system for linux and osx. if command is OK we pass mac, else always using linux.... 
here is some command for testing from web... need to tests it, bcz i do not have MAC ((
`say $(sw_vers -productName && sw_vers -productVersion \| sed 's/10//')`

File: .travis.yml [hunk 2]
```diff
@@ -28,39 +28,47 @@ matrix:
         - qmake
         - make install INSTALL_ROOT=$WORKDIR/build
         - androiddeployqt --output $WORKDIR/build
+  - os: osx
+    script: 
+      - QT_CI_PACKAGES=qt.qt5.597.clang_64 PATH=${TRAVIS_BUILD_DIR}/recipes:${TRAVIS_BUILD_DIR}/bin:${PATH} install-qt-osx 5.9.7 && source qt-5.9.7.env && qmake -version
```
Comment (van9konst) [inline .travis.yml]: maybe use only one script for testing on mac and linux platforms?

**Review by van9konst — CHANGES_REQUESTED** [review]

**Comment by tsteven4** [comment]
The qt installer changes over time and we have had bugs in extract-qt-installer that only effected installation of newer qt archives or older qt archives.  By using the full range of LTS releases we are likely to cover a wide range of qt installer evolution, thus maintaining compatibility with any archive the user selects. 

Additionally Qt has inconsistent naming of packages and the .travis.yml shows these differences which can be helpful when trying to figure out how to run install-qt on a particular archive.

So I would recommend we test all archives for LTS releases 5.6, 5.9, 5.12.   The tests are quick for linux, not so fast for osx.

File: .travis.yml (hunk 2 above)
Comment (tsteven4) [reply .travis.yml]: I agree they could be combined but I haven't chased that.  I also don't have a mac so all of this is only being run on travis which makes development slow.

Can we agree on the spelling of QT_TARGET_CATALOG vs. QT_target_catalog? Or should it really be QT_CI_TARGET_CATALOG?

File: recipes/install-qt-osx (hunk 1 above)
Comment (tsteven4) [reply recipes/install-qt-osx]: I pushed another commit that combines intall-qt and install-qt-osx.  I used uname to automatically detect Linux vs. Darwin and do the appropriate thing.

**Comment by tsteven4** [comment]
Note https://github.com/benlau/qtci/pull/19/commits/c73ca2a2f76d4d8a738a16374ba1be855926f32f fixes errors in creating the env file and plugs holes in regression in addition to combining install-qt and install-qt-osx.  See the commit message for more details.

**Comment by benlau** [comment]
hi @tsteven4

Thanks!

**Pull request merged**